backend/pkg/rest/applications_resources_controller.go

```diff
@@ -71,3 +73,150 @@ func (s *Server) GetApplicationResources(params operations.GetApplicationResourc
 			Total: &total,
 		})
 }
+
+func (s *Server) DeleteApplicationResource(id string) middleware.Responder {
```

Comment:
are you sure that you want to return in every failure here and leave half dead things behind? maybe it's better to delete as much as we can?

Reply:
Deleting objects should be in this order because for example if we cannot delete relationships, later we won't be able to delete the objects due to foreign keys in the join tables. Each operation works properly in that case the previous one succeeded.